lib/fluent/engine.rb

```diff
@@ -209,11 +209,11 @@ def run
         $log.error_backtrace
       ensure
         $log.info "shutting down fluentd"
-        shutdown
         if @log_emit_thread
           @log_event_loop_stop = true
           @log_emit_thread.join
         end
+        shutdown
```

Comment:
Yes. I wanted to make sure that the output plugin capturing events isn't terminated when log_emit_thread emit events. It seems to cause non-deterministic behaviour. But I'll try to capture more events event in `shutdown`.